src/test/java/com/liveramp/daemon_lib/built_in/TestCompositeDeserializer.java

```diff
@@ -41,4 +41,25 @@ public void testFailsEntirelyIfNoSerializerWorks() {
     deser.apply(bytes);
   }
 
-}
\ No newline at end of file
+  @Test
+  public void testCompositeOfComposites() {
+    final Charset charset = Charset.forName("UTF-8");
+    final IDConfig expected = new IDConfig(1L);
+    final byte[] bytes = new Gson().toJson(expected).getBytes(charset);
+    final CompositeDeserializer<IDConfig> deser = new CompositeDeserializer<>(Lists.newArrayList(
+        b -> {
+          throw new RuntimeException();
+        },
+        b -> new Gson().fromJson(new String(bytes, charset), IDConfig.class),
+        b -> new IDConfig(1L)
+    ));
+    final CompositeDeserializer<IDConfig> compcomp = new CompositeDeserializer<IDConfig>(Lists.newArrayList(
+        b -> { throw new RuntimeException(); },
+        deser
+    ));
+
+    final IDConfig actual = compcomp.apply(bytes);
+    org.junit.Assert.assertEquals(expected.getId(), actual.getId());
```

Comment:
why the package?

Reply:
So that when someone else includes another implementation of `assertEquals` in the same file, they won't conflict. I'm kidding. It's because I forgot to hit `Alt`+`Enter` and select `Import statically` in my IDE.

Reply:
Fixed now.